src/com/epam/javatrainings/classdecomposition/persistance/Pizza.java

```diff
@@ -0,0 +1,68 @@
+package com.epam.javatrainings.classdecomposition.persistance;
+
+import java.util.List;
+
+public class Pizza {
+    private String name;
+    private String type = "regular";
+    private List<Ingredient> ingredients;
+    private int quantity = 1;
+    private double pizzaBaseCost = 1;
+
+    public Pizza(String name, String type, int quantity) {
+        this.setName(name);
+        this.setType(type);
+        this.setQuantity(quantity);
+    }
+
+    public Pizza() {
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        if (name.length() >= 4 && name.length() <= 20 && name.matches("^[a-zA-Z ]+")) {
+            this.name = name;
+        }
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        if ("closed".equals(type)) {
```

Comment:
why we need this check ?

Reply:
This validations concept is that if client decides to order 'closed' type of pizza he/she should give the correct type, otherwise it will be considered that he/she has ordered 'regular' default type of pizza